cfpb/cfgov-refresh
# Careers: From Django you were born; to Django you shall return

Between the move to Sheer(like) and details missed during the transition, jobs/careers was in a bad state:

- We had never coded for the nuance of jobs overriding the salary range associated with their grade (I spent time looking at the processor and the API code-- I couldn't figure out where this logic was/should go)
- The WYSIWYG editor for job descriptions was gone
- We never came up with a reasonable way for non-devs to trigger 'sheer_index'

This pull request deals with these issues by importing a modified version of the old 'jobmanager' code into our codebase, using our new templates (with some tweaks for dealing with Django objects instead of elasticsearch results). Along the way, I added a static file that django-tinymce needs to function, and fixed the logic around salaries.

## Additions

- listed above

## Removals

- the 'careers' entry has been removed from SHEER_PROCESSORS


## Testing

- Delete the careers data from elasticsearch, you won't need that where we're going: `curl -XDELETE  http://localhost:9200/content/careers`
- Download and unzip [jobs.sql.zip](https://github.com/cfpb/cfgov-refresh/files/236773/jobs.sql.zip), and use you favorite method of loading an SQL file into your database (I tend to `cfgov/manage.py dbshell` and then `source path/to/jobs.sql`)
- make sure the old version of jobmanager isn't creeping in your virtualenv, `pip freeze | grep jobmanager` to see, `pip uninstall jobmanager` to get rid of it.
- -clone private repo CFPB/django-cfgov-common
- `pip install -e ../django-cfgov-common`
- `cfgov/manager.py runserver` and visit http://localhost:8000/about-us/careers and see that everything works the way you expect. Salaries should generally NOT match what's on www, as the grade salaries have been overridden in the Jobs data.
- Log in to http://localhost:8000/django-admin and look at the Jobs there. Observe that the wysisyg controls work in Job descriptions, and that you can edit the salary range min and max, and see the change at http://localhost:8000/about-us/careers/current-openings/
- Observe that you can blank out the salary range min and max, and see the page fall back to the min and max defined for the grade


## Review

- @richaagarwal @kave @kurtw 

-

## Todos

-

## Checklist
